.github/workflows/validate-case-studies-structures.yaml

```diff
@@ -0,0 +1,46 @@
+name: Validate case studies structures
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, ready_for_review]
+    paths:
+      - 'config/casestudies/*.yml'
+
+jobs:
+  validate:
+    if: github.event.pull_request.draft == false
+    name: Validate
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install ajv and yaml
+        run: npm install ajv@8.12.0 yaml@2.2.1
+      - name: Run validation script
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { readFile, readdir } = require("fs/promises");
+            const Ajv = require("ajv");
+            const YAML = require("yaml");
+
+            const ajv = new Ajv({
+              allErrors: true,
+            });
+
+            const schema = await readFile(".github/workflows/case-studies/schema.json", "utf-8");
+            const fileNames = await readdir("config/casestudies");
+
+            const validate = ajv.compile(JSON.parse(schema));
+
+            for (const fileName of fileNames) {
+              const study = await readFile(`config/casestudies/${fileName}`, "utf-8");
+              const valid = validate(YAML.parse(study));
+
+              if (valid === true) {
+                core.info(`${fileName} is valid`);
+              } else {
+                core.setFailed(`${fileName} is invalid:`);
+                core.setFailed(validate.errors);
+              }
+            }
```

Comment:
The whole working of the script should be part of try/catch block so as to handle the error properly and we can get proper console messages in the log for respective error.

Reply:
[`ajv.validate`](https://ajv.js.org/api.html#ajv-validate-schemaorref-object-string-data-any-boolean) does not throw an error in case of a false validation, it will simply return false.

Reply:
What if someone puts wrong schema in the file. Then `ajv.compile` will get into some errors and you are not handling that. So, please make sure you cover the script under try/catch block to offer good error handling.